tools/pubsys/src/aws/ami/mod.rs

```diff
@@ -382,13 +388,24 @@ async fn _run(args: &Args, ami_args: &AmiArgs) -> Result<HashMap<String, Image>>
         }
 
         let ec2_client = &ec2_clients[&region];
+
         let base_region = base_region.to_owned();
         let copy_future = ec2_client
             .copy_image()
             .set_description(tentative_amispec.description.clone())
             .set_name(Some(tentative_amispec.name.clone()))
             .set_source_image_id(Some(ids_of_image.image_id.clone()))
             .set_source_region(Some(base_region.as_ref().to_string()))
+            .set_tag_specifications(tentative_amispec.tags.as_ref().map(|x| {
+                vec![TagSpecification::builder()
+                    .resource_type(ResourceType::Image)
+                    .set_tags(Some(
+                        x.iter()
+                            .map(|(key, value)| Tag::builder().key(key).value(value).build())
+                            .collect(),
+                    ))
+                    .build()]
+            }))
```

Comment:
@jmt pointed out that despite my other comments being true about the initial registration, we do still need to add code to handle tag copies during the image copy.

I think rather than manually inserting them based on the tentative ami spec (which would miss any additional logic handled during the rendering process) we can call `set_copy_image_tags()`